core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -1893,6 +1899,39 @@ private static class TimestampAddConvertlet implements SqlRexConvertlet {
     }
   }
 
+  /** Convertlet that handles the BigQuery {@code TIMESTAMP_SUB} function. */
+  private static class TimestampSubConvertlet implements SqlRexConvertlet {
+    @Override public RexNode convertCall(SqlRexContext cx, SqlCall call) {
+      // TIMESTAMP_SUB(timestamp, interval)
+      //  => timestamp - count * INTERVAL '1' UNIT
+      final RexBuilder rexBuilder = cx.getRexBuilder();
+      final SqlBasicCall operandCall = call.operand(1);
+      SqlIntervalQualifier qualifier = operandCall.operand(1);
+      final RexNode op1 = cx.convertExpression(operandCall.operand(0));
+      final RexNode op2 = cx.convertExpression(call.operand(0));
+      final TimeFrame timeFrame = cx.getValidator().validateTimeFrame(qualifier);
+      final TimeUnit unit = timeFrame.unit();
+      RexNode interval2Sub;
```

Comment:
`interval2Sub` should be `final`.

Fix the checkerframework errors.

Add comment why special handling for NANO, MICRO